src/Microsoft.Diagnostics.Monitoring/AggregateServiceProvider.cs

```diff
@@ -0,0 +1,35 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System;
+using System.Collections.Generic;
+using System.Text;
+
+namespace Microsoft.Diagnostics.Monitoring
+{
+    /// <summary>
+    /// Allows chaining of service providers.
+    /// </summary>
+    public class AggregateServiceProvider : IServiceProvider
```

Comment:
This is why you shouldn't have 2 service providers, can we clean this up now? One service provider and avoid this completely.

Reply:
Please see the other comment thread. I believe I can remove this class, but I don't see a way to have only 1 service container while maintaining ILogger as our interface for log sinks.